includes/wc-template-functions.php

```diff
@@ -2728,3 +2728,41 @@ function wc_page_noindex() {
 	}
 }
 add_action( 'wp_head', 'wc_page_noindex' );
+
+/**
+ * Get a slug identifying the current theme.
+ *
+ * @since 3.3.0
+ * @return string
+ */
+function wc_get_theme_slug_for_templates() {
+	/**
+	 * @hooked wc_check_for_underscores_theme
+	 * @since 3.3.0
+	 */
+	$template = apply_filters( 'woocommerce_theme_slug_for_templates', get_option( 'template' ) );
+
+	return $template;
+}
+
+/**
+ * Treat all themes that don't formally support WooCommerce and are based on Underscores as one 'underscores' theme for templates.
+ *
+ * @since 3.3.0
+ * @param string $template
+ * @return string
+ */
+function wc_check_for_underscores_theme( $template ) {
```

Comment:
Thats a lot of work to check if a theme is maybe underscores :( Maybe we should look into the original suggestion - I think it's unlikely many themes are based on 2010 and don't use their own wrappers, so we could default to underscores and add a wrapper specifically for 2010. Underscores will have more usage than 2010 - even some other default wp themes (2016) use underscores style wrappers, and it negates the need for this function.